test/scripts/run_test.py

```diff
@@ -33,42 +44,54 @@ def analyse_result(result):
     return ok_num
 
 
-def run_cases(module, cases, log_dir, device):
-    print '\n> running cases on device', device
-    result_file = os.path.join(log_dir, module + '-result-' + LOGCAT_SUFFIX)
-    logcat_file = os.path.join(log_dir, module + '-logcat-' + LOGCAT_SUFFIX)
-    for case in cases:
-        adb = ['adb'] if device == None else ['adb', '-s', device]
-        if module == 'base':
-            target_package = BASE_TARGET_PACKAGE
-        elif module == 'conference':
-            target_package = CONF_TARGET_PACKAGE
-        elif module == 'p2p':
-            target_package = P2P_TARGET_PACKAGE
-        clean_logcat = ['logcat', '-c']
-        subprocess.call(adb + clean_logcat)
-        am_cmd = ['shell', 'am', 'instrument', '-w', '-r', '-e', 'debug', 'false', '-e',
-                  'class', target_package + '.' + case,
-                  target_package + '.test/android.test.InstrumentationTestRunner']
-        with open(result_file, 'a+') as rf:
-            subprocess.call(adb + am_cmd, stdout=rf)
-        logcat_cmd = ['logcat', '-d', target_package]
-        with open(logcat_file, 'a+') as lf:
-            subprocess.call(adb + logcat_cmd, stdout=lf)
+def analyse_unit_test_result(result):
+    total_num = 0
+    fail_num = 0
+    with open(result, 'r') as f:
+        for line in f:
+            m_failure = re.match('Tests run:\s+(\d+).*?Failures:\s+(\d+)', line)
+            m_succeed = re.match('OK.*?(\d+) test', line)
+            if m_failure is not None:
+                total_num = int(m_failure.group(1))
+                fail_num = int(m_failure.group(2))
+            if m_succeed is not None:
+                total_num = int(m_succeed.group(1))
+    return total_num, fail_num
```

Comment:
Actually `INSTRUMENTATION_STATUS_CODE` can be and better be used to calculate the number and the result of test cases.
I'm ok with the current code tho, but please leave a //TODO to change later

Reply:
`// TODO: merge analyse_unit_test_result and analyse_instrumentation_test_result with checking  INSTRUMENTATION_STATUS_CODE of each test case.`

Reply:
done

Reply:
In the situation of building failure, `total_num`, `fail_num`, `succeed` will all be 0, will result in the success of running unit test.
This is an incorrect behavior.

Reply:
`total_num` means the number of cases have been executed or the number of cases defined in the tests?

Reply:
total_num  means the number of cases have been executed

Reply:
but actual all defined cases will be executed

Reply:
but actually all defined cases will be executed

Reply:
No i don't really think so. `total_num` will be 0 upon building failure.

Reply:
if build failed cases can not executed , so ,if total_num = 0 building will be failure, other wise total_num !=0